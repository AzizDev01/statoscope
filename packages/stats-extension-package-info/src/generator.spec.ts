import { ExtensionDescriptor } from '@statoscope/stats/spec/extension';
import Generator from './generator';

const adapter: ExtensionDescriptor = { name: 'test-adapter', version: '7.7.7' };

test('should work', () => {
  const generator = new Generator(adapter);

  generator.handleInstance('foo-compilation', 'foo-package', 'foo-instance', {
    version: '1.0.0',
  });
  generator.handleInstance('foo-compilation', 'foo-package', 'bar-instance', {
    version: '2.0.0',
  });
  generator.handleInstance('foo-compilation', 'bar-package', 'bar-instance', {
    version: '3.0.0',
  });
  generator.handleInstance('bar-compilation', 'foo-package', 'bar-instance', {
    version: '4.0.0',
  });

  expect(generator.get()).toMatchSnapshot();
});
