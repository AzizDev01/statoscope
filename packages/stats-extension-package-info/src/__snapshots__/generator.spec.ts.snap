// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should work 1`] = `
Object {
  "descriptor": Object {
    "adapter": Object {
      "name": "test-adapter",
      "version": "7.7.7",
    },
    "author": Object {
      "name": "Sergey Melyukov",
      "url": "https://twitter.com/smelukov",
    },
    "homepage": "https://github.com/statoscope/statoscope/",
    "name": "@statoscope/stats-extension-compressed",
    "version": "5.2.0",
  },
  "payload": Object {
    "compilations": Array [
      Object {
        "id": "foo-compilation",
        "packages": Array [
          Object {
            "instances": Array [
              Object {
                "info": Object {
                  "version": "1.0.0",
                },
                "path": "foo-instance",
              },
              Object {
                "info": Object {
                  "version": "2.0.0",
                },
                "path": "bar-instance",
              },
            ],
            "name": "foo-package",
          },
          Object {
            "instances": Array [
              Object {
                "info": Object {
                  "version": "3.0.0",
                },
                "path": "bar-instance",
              },
            ],
            "name": "bar-package",
          },
        ],
      },
      Object {
        "id": "bar-compilation",
        "packages": Array [
          Object {
            "instances": Array [
              Object {
                "info": Object {
                  "version": "4.0.0",
                },
                "path": "bar-instance",
              },
            ],
            "name": "foo-package",
          },
        ],
      },
    ],
  },
}
`;
